ladybug/windrose.py

```diff
@@ -195,56 +195,68 @@ def frequency_spacing_distance(self):
         return self._frequency_spacing_distance
 
     @frequency_spacing_distance.setter
-    def frequency_spacing_distance(self, frequency_spacing_distance):
-        f = frequency_spacing_distance
-        assert f > 0.0 and isinstance(f, (float, int)), 'frequency_spacing_distance' \
-            'should be a number greater then 0. Got {}.'.format(f)
+    def frequency_spacing_distance(self, val):
+        assert val > 0.0 and isinstance(val, (float, int)), \
+            'frequency_spacing_distance should be a number greater then 0. ' \
+            'Got {}.'.format(val)
         self._compass = None
         self._container = None
-        self._frequency_spacing_distance = frequency_spacing_distance
+        self._frequency_spacing_distance = val
 
     @property
+    def frequency_spacing_hypot_distance(self):
+        """Length of radius of windrose assuming perpendicular frequency spacing.
+
+        The limit of the frequency_spacing_hypot_distance as number of directions
+        approaches infinity equals the frequency_spacing_distance (as is the case
+        with a circle).
+        """
+        _theta = 360.0 / self._number_of_directions
+        theta = _theta / 2.0 / 180.0 * math.pi
+        return self.frequency_spacing_distance / math.cos(theta)
+
+    @ property
     def frequency_maximum(self):
         """Get maximum frequency to represent in the windrose plot.
 
         Default: number of items in largest windrose histogram bin.
         """
         return self.frequency_intervals_compass * self.frequency_hours
 
-    @property
+    @ property
     def frequency_hours(self):
         """Get or set number of hours to bin analysis values, per direction."""
         if self._frequency_hours is None:
             self._frequency_hours = self.DEFAULT_FREQUENCY_HOURS
         return self._frequency_hours
 
-    @frequency_hours.setter
+    @ frequency_hours.setter
     def frequency_hours(self, frequency_hours):
         assert frequency_hours > 0, 'frequency_hours must be a number greater then 0. ' \
             'Got: {}'.format(frequency_hours)
         self._frequency_hours = int(frequency_hours)
 
-    @property
+    @ property
     def frequency_intervals_compass(self):
         """Get or set the number of intervals in the windrose compass."""
         if self._frequency_intervals_compass is None:
             self._frequency_intervals_compass = self.frequency_intervals_mesh
         return self._frequency_intervals_compass
 
-    @frequency_intervals_compass.setter
+    @ frequency_intervals_compass.setter
     def frequency_intervals_compass(self, frequency_intervals_compass):
-        assert frequency_intervals_compass >= 1, 'The frequency_intervals_compass ' \
-            'must be greater then 0. Got: {}'.format(frequency_intervals_compass)
+        assert frequency_intervals_compass >= 1, 'The frequency_intervals_compass '
+        'must be greater then 0. Got: {}'.format(frequency_intervals_compass)
```

Comment:
What you had before was correct. Now, people won't get the full assertion message in the event that this is triggered. See here for an example:
![image](https://user-images.githubusercontent.com/5567574/97733602-75ec8a80-1aae-11eb-992f-bc11bb9551e7.png)